src/pymatgen/io/vasp/inputs.py

```diff
@@ -424,8 +435,12 @@ def from_str(
                 for idx, n_atom in enumerate(n_atoms):
                     atomic_symbols.extend([default_names[idx]] * n_atom)
                 vasp5_symbols = True
-            except IndexError:
-                pass
+            except IndexError as exc:
+                # TODO: need discussion on exception handling
+                warnings.warn(
+                    f"Elements in POTCAR {default_names} don't match POSCAR {symbols}", BadPoscarWarning, stacklevel=2
+                )
+                raise ValueError(f"Elements in POTCAR {default_names} don't match POSCAR {symbols}") from exc
```

Comment:
Is there any use case the need this `IndexError` being suppressed instead of interrupted?